lib/source-map-resolve-node.js

```diff
@@ -71,8 +71,45 @@ function resolveSourceMapSync(code, codeUrl, read) {
 }
 
 var dataUriRegex = /^data:([^,;]*)(;[^,;]*)*(?:,(.*))?$/
+
+/**
+ * The media type for JSON text is application/json.
+ * 
+ * {@link https://tools.ietf.org/html/rfc8259#section-11 | IANA Considerations }
+ * 
+ * `text/json` is non-standard media type
+ */
 var jsonMimeTypeRegex = /^(?:application|text)\/json$/
 
+/**
+ * JSON text exchanged between systems that are not part of a closed ecosystem
+ * MUST be encoded using UTF-8.
+ * 
+ * {@link https://tools.ietf.org/html/rfc8259#section-8.1 | Character Encoding}
+ */
+var jsonCharacterEncoding = "utf-8"
+
+function base64ToBuf(b64) {
+  var binStr = atob(b64)
+  var len = binStr.length
+  var arr = new Uint8Array(len)
+  for (var i = 0; i < len; i++) {
+    arr[i] = binStr.charCodeAt(i)
+  }
+  return arr
+}
+
+function decodeBase64String(b64) {
+  if (typeof TextDecoder === "undefined" || typeof Uint8Array === "undefined") {
+    return atob(b64)
+  }
+  var buf = base64ToBuf(b64);
+  // Note: `decoder.decode` method will throw a `DOMException` with the
+  // `"EncodingError"` value when an coding error is found.
```

Comment:
I need to catch this error and add a test for it.

Reply:
Or maybe it’s better to use `fatal: false`?

Reply:
@xiaoxiangmoe what do you think? Are utf8 errors commonly generated by tools? What is more helpful to the user – errors or replacement characters?

Reply:
I think `fatal: true` is better for users. It doesn't happen very often, but once it does, it's easier to spot the problem if an error is throw.

Reply:
Ok. Do you know how I can create a base64 with a utf8 error in it, so I can write a test?

Reply:
We might construct an lantin1-encoded base64(with some non ascii char in 0xA0-0xFF), then read it in utf8, and it may raise a utf8 error.

Reply:
I'm sorry, my computer is out of power, I can't easily test it with my mobile phone now. We might find a typical illegal utf-8 sequence and write the test as lantin 1

Reply:
https://stackoverflow.com/questions/1301402/example-invalid-utf8-string

Reply:
I found through trial and error that `abc` results in invalid utf8. 🤷‍♂